### PR TITLE
all is done

### PR DESCRIPTION
**Pull request by ivlytvynets**

### CODE REVIEW

**Review by gigbat — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1 +1,51 @@
...
+* Controller layer
+
+Communication between back-end and front-end is performed via DTO and JSON.
+## Available functionality
+
+### User
+* Register in the system
+* Log in to the system
+* Browse available topics
+* Browse available intensive courses
```
Comment (KseniiaMakarova) [inline README.md]: what topics and courses?

File: README.md [hunk 2]
```diff
@@ -1 +1,51 @@
...
+## Requirements for launching the application
+
+1. Download and install JDK 8 or later
+2. Download and install Apache Tomcat web server
+3. Download and install MySQL.
+4. Configure the database connection properties in the **db.properties** file
+* user: "your username"
+* password: "your password"
+* db.url=jdbc:mysql://localhost/*<your_database>*?currentSchema=*<your_schema>*
+5. Run a project using the IDE
```
Comment (KseniiaMakarova) [inline README.md:51]: is admin injected in your app? how can I test the app with ADMIN role?

File: src/main/java/com/dev/theatre/controller/PerformanceController.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movies")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/controller/PerformanceController.java]: rename mapping

File: src/main/java/com/dev/theatre/config/SecurityConfig.java [hunk 4]
```diff
@@ -33,7 +33,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 .authorizeRequests()
                 .antMatchers("/register").permitAll()
                 .antMatchers(HttpMethod.POST, "/movies/**").hasRole("ADMIN")
-                .antMatchers(HttpMethod.POST, "/cinema-halls/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.POST, "/theatre-halls/**").hasRole("ADMIN")
                 .antMatchers(HttpMethod.POST, "/movie-sessions/**").hasRole("ADMIN")
                 .antMatchers(HttpMethod.PUT, "/movie-sessions/**").hasRole("ADMIN")
                 .antMatchers(HttpMethod.DELETE, "/movie-sessions/**").hasRole("ADMIN")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/config/SecurityConfig.java]: rename this

File: src/main/java/com/dev/theatre/controller/PerformanceSessionController.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/controller/PerformanceSessionController.java]: same

File: src/main/java/com/dev/theatre/controller/ShoppingCartController.java [hunk 6]
```diff
@@ -19,25 +19,25 @@
...
                                   UserService userService,
                                   ShoppingCartMapper shoppingCartMapper) {
         this.shoppingCartService = shoppingCartService;
-        this.movieSessionService = movieSessionService;
+        this.performanceSessionService = performanceSessionService;
         this.userService = userService;
         this.shoppingCartMapper = shoppingCartMapper;
     }
 
     @PostMapping("/movie-sessions")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/controller/ShoppingCartController.java]: same

File: src/main/java/com/dev/theatre/controller/TheatreStageController.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/controller/TheatreStageController.java]: same

File: src/main/java/com/dev/theatre/dao/impl/TheatreStageDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+            return theatreStages;
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get all theatre halls", e);
+        }
+    }
+
+    @Override
+    public Optional<TheatreStage> get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.load(TheatreStage.class, id));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/dao/impl/TheatreStageDaoImpl.java]:
```suggestion
            return Optional.ofNullable(session.get(TheatreStage.class, id));
```

File: src/main/java/com/dev/theatre/model/PerformanceSession.java [hunk 9]
```diff
@@ -39,29 +39,29 @@ public void setShowTime(LocalDateTime showTime) {
         this.showTime = showTime;
     }
 
-    public Movie getMovie() {
-        return movie;
+    public Performance getMovie() {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/model/PerformanceSession.java]: rename getters and setters

File: src/main/java/com/dev/theatre/model/Ticket.java [hunk 10]
```diff
@@ -26,12 +26,12 @@ public void setId(Long id) {
         this.id = id;
     }
 
-    public MovieSession getMovieSession() {
-        return movieSession;
+    public PerformanceSession getMovieSession() {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/model/Ticket.java]: same

File: src/main/java/com/dev/theatre/model/dto/PerformanceSessionRequestDto.java [hunk 11]
```diff
@@ -1,9 +1,9 @@
-package com.dev.cinema.model.dto;
+package com.dev.theatre.model.dto;
 
 import jakarta.validation.constraints.NotNull;
 import java.time.LocalDateTime;
 
-public class MovieSessionRequestDto {
+public class PerformanceSessionRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/model/dto/PerformanceSessionRequestDto.java:6]: review all fields

File: src/main/java/com/dev/theatre/model/dto/PerformanceSessionResponseDto.java [hunk 12]
```diff
@@ -1,8 +1,8 @@
-package com.dev.cinema.model.dto;
+package com.dev.theatre.model.dto;
 
 import java.time.LocalDateTime;
 
-public class MovieSessionResponseDto {
+public class PerformanceSessionResponseDto {
     private Long movieSessionId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/model/dto/PerformanceSessionResponseDto.java]: same

File: src/main/java/com/dev/theatre/service/impl/PerformanceServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Override
+    public List<Performance> getAll() {
+        return performanceDao.getAll();
+    }
+
+    @Override
+    public Performance get(Long id) {
+        return performanceDao.get(id).orElseThrow(
+                () -> new DataProcessingException("Can't get movie with id: " + id));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/service/impl/PerformanceServiceImpl.java]: rename this

File: src/main/java/com/dev/theatre/service/impl/PerformanceSessionMapperImpl.java [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+                                        TheatreStageService theatreStageService) {
+        this.performanceService = performanceService;
+        this.theatreStageService = theatreStageService;
+    }
+
+    @Override
+    public PerformanceSessionResponseDto getResponseDto(PerformanceSession performanceSession) {
+        PerformanceSessionResponseDto performanceSessionResponseDto =
+                new PerformanceSessionResponseDto();
+        performanceSessionResponseDto.setMovieSessionId(performanceSession.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/service/impl/PerformanceSessionMapperImpl.java]: same

File: src/main/java/com/dev/theatre/service/impl/PerformanceSessionMapperImpl.java [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+        performanceSessionResponseDto.setMovieTitle(performanceSession.getMovie().getTitle());
+        performanceSessionResponseDto.setCinemaHallId(performanceSession.getCinemaHall().getId());
+        return performanceSessionResponseDto;
+    }
+
+    @Override
+    public PerformanceSession getEntity(PerformanceSessionRequestDto performanceSessionRequestDto) {
+        PerformanceSession performanceSession = new PerformanceSession();
+        performanceSession.setMovie(performanceService.get(performanceSessionRequestDto
+                .getMovieId()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/service/impl/PerformanceSessionMapperImpl.java]: same

File: src/main/java/com/dev/theatre/service/impl/PerformanceSessionServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public PerformanceSession update(PerformanceSession performanceSession) {
+        return performanceSessionDao.update(performanceSession);
+    }
+
+    @Override
+    public PerformanceSession get(Long id) {
+        return performanceSessionDao.get(id).orElseThrow(() ->
+                new DataProcessingException("Can't get movie session with id: " + id));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/service/impl/PerformanceSessionServiceImpl.java]: same

File: src/main/java/com/dev/theatre/service/impl/ShoppingCartServiceImpl.java [hunk 17]
```diff
@@ -20,9 +20,9 @@ public ShoppingCartServiceImpl(ShoppingCartDao shoppingCartDao, TicketDao ticket
     }
 
     @Override
-    public void addSession(MovieSession movieSession, User user) {
+    public void addSession(PerformanceSession performanceSession, User user) {
         Ticket ticket = new Ticket();
-        ticket.setMovieSession(movieSession);
+        ticket.setMovieSession(performanceSession);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/service/impl/ShoppingCartServiceImpl.java]: same

File: src/main/java/com/dev/theatre/service/impl/TheatreStageServiceImpl.java [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Override
+    public List<TheatreStage> getAll() {
+        return theatreStageDao.getAll();
+    }
+
+    @Override
+    public TheatreStage get(Long id) {
+        return theatreStageDao.get(id).orElseThrow(
+                () -> new DataProcessingException("Can't get theatre hall with id: " + id));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/service/impl/TheatreStageServiceImpl.java]:
```suggestion
                () -> new DataProcessingException("Can't get theatre stage with id: " + id));
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
remove redundant branches from GitHub and edit `groupId` and `artifactId` in pom.xml

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
please also see comments that were hidden

File: src/main/java/com/dev/theatre/config/SecurityConfig.java [hunk 19]
```diff
@@ -32,11 +32,10 @@ protected void configure(HttpSecurity http) throws Exception {
                 .and()
                 .authorizeRequests()
                 .antMatchers("/register").permitAll()
-                .antMatchers(HttpMethod.POST, "/movies/**").hasRole("ADMIN")
-                .antMatchers(HttpMethod.POST, "/theatre-halls/**").hasRole("ADMIN")
-                .antMatchers(HttpMethod.POST, "/movie-sessions/**").hasRole("ADMIN")
-                .antMatchers(HttpMethod.PUT, "/movie-sessions/**").hasRole("ADMIN")
-                .antMatchers(HttpMethod.DELETE, "/movie-sessions/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.POST, "/performances/**").hasRole("ADMIN")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/config/SecurityConfig.java:35]: what about theatre-stages?

File: src/main/java/com/dev/theatre/model/Ticket.java [hunk 20]
```diff
@@ -30,7 +30,7 @@ public PerformanceSession getMovieSession() {
         return performanceSession;
     }
 
-    public void setMovieSession(PerformanceSession performanceSession) {
+    public void setPerformanceSession(PerformanceSession performanceSession) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/model/Ticket.java:33]: what about getter?

File: src/main/java/com/dev/theatre/service/impl/PerformanceSessionServiceImpl.java [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+@Service
+public class PerformanceSessionServiceImpl implements PerformanceSessionService {
+    private final PerformanceSessionDao performanceSessionDao;
+
+    public PerformanceSessionServiceImpl(PerformanceSessionDao performanceSessionDao) {
+        this.performanceSessionDao = performanceSessionDao;
+    }
+
+    @Override
+    public List<PerformanceSession> findAvailableSessions(Long movieId, LocalDate date) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theatre/service/impl/PerformanceSessionServiceImpl.java]: rename parameter

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**